server/src/main/java/org/candlepin/model/Product.java

```diff
@@ -1269,4 +1308,44 @@ public void updateEntityVersion() {
         this.entityVersion = this.getEntityVersion();
     }
 
+    /**
+     * Retrieves a collection of provided product for this product.
+     *
+     * @return returns the provided product of this product.
+     */
+    @JsonIgnore
+    public Collection<Product> getProvidedProducts() {
+        return new SetView<>(this.providedProducts);
+    }
+
+    /**
+     * Method to set provided products.
+     *
+     * @param providedProducts A collection of provided products.
+     * @return A reference to this product.
+     */
+    public Product setProvidedProducts(Collection<Product> providedProducts) {
+
+        if (providedProducts != null) {
+            this.providedProducts = new HashSet<>(providedProducts);
+        }
+        else {
+            this.providedProducts = new HashSet<>();
+        }
+
+        return this;
+    }
+
+    /**
+     * Adds the specified provided product as one of the provided product of this product. If the provided
+     * product is already there, it will not be added again.
+     *
+     * @param provided provided product
```

Comment:
Try to be a little clearer with the parameter comments, even if it means being a bit verbose or repetitive. For other readers and code documentation processors, there may be times when the visible part of the documentation is the parameter's text rather than the method's doc block.

Reply:
Fixed it. Thanks